### PR TITLE
V415

### LINKED ISSUE #121: Validation (2017): V415 (Preapproval)
**Edit #:** V415
**Type:** Validity
**Category:** Loan/Application Register (only)
**Column:** Preapproval

Preapproval must = 1, 2, or 3.

Additional resources:
* [FFIEC's 2016 HMDA File Spec](https://www.ffiec.gov/hmda/pdf/spec2016.pdf)
* [FFIEC's 2016 HMDA Edits](https://www.ffiec.gov/hmda/pdf/edit2016.pdf)
* [CFPB's 2017 HMDA File Spec](http://www.consumerfinance.gov/hmda/static/for-filers/2017/2017-HMDA-File-Specifications.pdf)

### PR DESCRIPTION
**Pull request by kgudel**
Closes #121

### CODE REVIEW

File: validation/src/main/scala/hmda/validation/rules/lar/validity/V415.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+object V415 extends EditCheck[LoanApplicationRegister] {
+
+  val preApprovalTypes = List(1, 2, 3)
+
+  def apply(lar: LoanApplicationRegister): Result = {
+    lar.preapprovals is containedIn(preApprovalTypes)
+  }
+
+  def name = "V415"
+}
```
Comment (jmarin) [inline validation/src/main/scala/hmda/validation/rules/lar/validity/V415.scala]: `apply` and `name` need `override` modifier

File: validation/src/test/scala/hmda/validation/rules/lar/validity/V415Spec.scala [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+class V415Spec extends LarEditCheckSpec with BadValueUtils {
+  property("Preapprovals must be 1, 2, or 3") {
+    forAll(larGen) { lar =>
+      lar.mustPass
+    }
+  }
+
+  val invalidPreaaprovalsGen: Gen[Int] = intOutsideRange(1, 3)
+
```
Comment (jmarin) [inline validation/src/test/scala/hmda/validation/rules/lar/validity/V415Spec.scala]: Not a big issue, but I think there is an extra `a` in this variable name

File: validation/src/main/scala/hmda/validation/rules/lar/validity/V415.scala [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package hmda.validation.rules.lar.validity
+
+import hmda.model.fi.lar.LoanApplicationRegister
+import hmda.validation.dsl.Result
+import hmda.validation.rules.EditCheck
+
+object V415 extends EditCheck[LoanApplicationRegister] {
+
+  val preApprovalTypes = List(1, 2, 3)
```
Comment (schbetsy) [inline validation/src/main/scala/hmda/validation/rules/lar/validity/V415.scala]: According to the File Specification, these values stand for:
```
1. Preapproval was requested
2. Preapproval was not requested
3. Not applicable
```

I wouldn't call them preApproval "types", per se. Consider renaming this variable or removing it.

**Comment by nickgrippin** [comment]
LGTM

**Pull request merged**